libs/wlgen/wlgen/workload.py

```diff
@@ -193,25 +199,26 @@ def run(self,
 
         self.cgroup = cgroup
 
-        if self.command is None:
+        # Compose the actual execution command starting from the base command
+        # defined by the base class
+        _command = self.command
+
+        if not _command:
             logging.error('%14s - Error: empty executor command', 'WlGen')
 
         # Prepend eventually required taskset command
-        if self.cpus:
-            cpus_mask = self.getCpusMask(self.cpus)
+        if cpus or self.cpus:
+            cpus_mask = self.getCpusMask(cpus if cpus else self.cpus)
             self.taskset_cmd = '{}/taskset 0x{:X}'\
                     .format(self.target.executables_directory,
                             cpus_mask)
-            self.command = '{} {}'\
-                    .format(self.taskset_cmd, self.command)
+            _command = '{} {}'\
+                    .format(self.taskset_cmd, _command)
 
         # Prepend eventually required cgroup command
-        if self.cgroup and self.cgroup_cmd == '':
-            self.cgroup_cmd = 'cgroups_run_into {}'\
-                .format(self.target.executables_directory,
-                        self.cgroup)
-            self.command = '{} \'{}\''\
-                .format(self.cgroup_cmd, self.command)
+        if self.cgroup:
+            self.cgroup_cmd = 'cgroups_run_into {}'.format(self.cgroup)
```

Comment:
This fails because `cgroups_run_into` expects a path to a Cgroup, whereas here it gets the object reference (i.e. `<devlib.module.cgroups.CGroup object at 0x7fcf9c747ad0>`).

You should change `self.cgroup` to `self.cgroup.name`.